Src/IronPython/Runtime/TypecodeOps.cs

```diff
@@ -0,0 +1,105 @@
+﻿using IronPython.Runtime.Operations;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace IronPython.Runtime {
+    class TypecodeOps {
+
+        public static bool IsTypecodeFormat(string format) {
+            switch (format) {
+                case "c": // char
+                case "b": // signed byte
+                case "B": // unsigned byte
+                case "x": // pad byte
+                case "s": // null-terminated string
+                case "p": // Pascal string
+                case "u": // unicode char
+                case "h": // signed short
+                case "H": // unsigned short
+                case "i": // signed int
+                case "I": // unsigned int
+                case "l": // signed long
+                case "L": // unsigned long
+                case "f": // float
+                case "P": // pointer
+                case "q": // signed long long
+                case "Q": // unsigned long long
+                case "d": // double
+                    return true;
+                default:
+                    return false;
+            }
+        }
+
+        public static int GetTypecodeWidth(char typecode) {
```

Comment:
Looking at usage it seems like you mostly do things like `if(IsTypecodeFormat(typecode)) GetTypecodeWidth(typecode)`. I wonder if using the Try pattern would be better suited `bool TryGetTypecodeWidth(typecode, out int width)`? Same comment for the other methods.